src/omero_zarr/raw_pixels.py

```diff
@@ -74,6 +75,83 @@ def planeGen() -> np.ndarray:
     print("Created", name)
 
 
+def add_image(image: omero.gateway.Image, parent: Group, field_index="0") -> None:
+    """Adds the image pixel data as array to the given parent zarr group."""
+    size_c = image.getSizeC()
+    size_z = image.getSizeZ()
+    size_x = image.getSizeX()
+    size_y = image.getSizeY()
+    size_t = image.getSizeT()
+    d_type = image.getPixelsType()
+
+    group = parent.create(
+        field_index,
+        shape=(size_t, size_c, size_z, size_y, size_x),
+        chunks=(1, 1, 1, size_y, size_x),
+        dtype=d_type,
+    )
+
+    zct_list = []
+    for t in range(size_t):
+        for c in range(size_c):
+            for z in range(size_z):
+                zct_list.append((z, c, t))
+
+    pixels = image.getPrimaryPixels()
+
+    def planeGen() -> np.ndarray:
+        planes = pixels.getPlanes(zct_list)
+        yield from planes
+
+    planes = planeGen()
+
+    for t in range(size_t):
+        for c in range(size_c):
+            for z in range(size_z):
+                plane = next(planes)
+                group[t, c, z, :, :] = plane
+
+
+def plate_to_zarr(plate: omero.gateway._PlateWrapper, args: argparse.Namespace) -> None:
+    """
+       Exports a plate to a zarr file using the hierarchy discussed here ('Option 3'):
+       https://github.com/ome/omero-ms-zarr/issues/73#issuecomment-706770955
+    """
+    gs = plate.getGridSize()
+    n_rows = gs["rows"]
+    n_cols = gs["columns"]
+    n_fields = plate.getNumberOfFields()
+    total = n_rows * n_cols * (n_fields[1] - n_fields[0] + 1)
```

Comment:
The actual number of Images may be smaller since not all Wells will have the max number of fields (or any fields).
But I guess that's and edge case and doesn't matter too much just for printing status.